src/main.js

```diff
@@ -0,0 +1,305 @@
+						context.commit('setApiKeyNeeded', false)
+					}
+
+				})
+		},
+		deleteCity(context, cityId) {
+			axios.post(generateUrl('/apps/weather/city/delete'), { id: cityId })
+				.then((response) => {
+					if (response.data != null && !!response.data.deleted) {
+						context.commit('removeCity', cityId)
+						// If current city is the removed city, close it
+						if (context.state.selectedCityId === cityId) {
+							context.commit('unsetSelected')
+							// if there are still cities on the list, and the selected was deleted, select the home city.
+							// if the home city was deleted, select the first one instead
+							if (context.state.allCities.cities.length > 0) {
+								context.dispatch(
+									'loadCity',
+									cityId !== context.state.allCities.home
+										? context.state.allCities.home
+										: context.state.allCities.cities[0].id)
+							}
+						}
+					} else {
+						alert(t('weather', 'Failed to remove city. Please contact your administrator'))
+					}
+				}).catch((reason) => {
+					showError(context.getters.fatalError)
+					logger.error('Error deleting city', { reason })
+				})
+		},
+		setHome(context, cityId) {
+			axios.post(generateUrl('/apps/weather/settings/home/set'), { city: cityId })
+				.then(response => {
+					if (response.data != null && !!response.data.set) {
+						context.commit('setHome', cityId)
+					} else {
+						showError(t('weather', 'Failed to set home. Please contact your administrator'))
+					}
+				})
+				.catch((reason) => {
+					logger.error('Error setting home city', { reason })
+					showError(context.getters.fatalError)
+				})
+		},
+		selectCity(context, cityId) {
+			context.dispatch('loadCity', cityId)
+				.then(() => {
+					context.commit('setSelected', cityId)
+				})
+		},
+		loadMetric(context) {
+			axios.get(generateUrl('/apps/weather/settings/metric/get'))
+				.then((result) => {
+					if (result.data.metric) {
+						context.commit('setMetric', result.data.metric)
+						context.commit('setMetricRepresentation', mapMetric(result.data.metric))
+					}
+				}).catch((reason) => {
+					logger.error('Error setting metric', { reason })
+					showError(context.getters.fatalError)
+				})
+		},
+		modifyMetric(context, metric) {
+			axios.post(generateUrl('/apps/weather/settings/metric/set'), { metric })
+				.then((result) => {
+					if (result.data != null && !!(result.data.set)) {
+						context.commit('setMetric', metric)
+						context.commit('setMetricRepresentation', mapMetric(metric))
+						context.dispatch('loadCity', context.state.selectedCityId)
+					} else {
+						showError(t('weather', 'Failed to set metric. Please contact your administrator'))
+					}
+				}).catch(
+					(reason) => {
+						if (reason.status === 404) {
+							showError(t('weather', 'This metric is not known.'))
+						} else {
+							logger.error('Error setting metric', { reason })
+							showError(context.getters.fatalError)
+						}
+					})
+		},
+	},
```

Comment:
```suggestion
		async addCity(context, newCityName) {
			try {
				const { data } = await axios.post(generateUrl('/apps/weather/city/add'), { name: newCityName })
				showSuccess(`City added: ${newCityName}`)
				if (data.id) {
					// response contains the id, but not the name
					context.commit('addCity', { name: newCityName, id: response.data.id })
				}
				// otherwise one may send a new API call : context.dispatch('loadCities')
			} catch (reason) {
				logger.error('Error adding city', { reason })
				showError(context.getters.fatalError)
			}
		},
		async loadCities(context) {
			const { data } = await axios.get(generateUrl('/apps/weather/city/getall'))
			try {
				if (data.home) {
					context.commit('setHome', data.home)
				}
				if (data.cities) {
					context.commit('setCities', data.cities)
				}
				if (!context.state.loaded) {
					if (data.home && !!data.cities.find(city => city.id === data.home)) {
						// there is a home city set, and it hasnn't been removed before -> load it
						context.dispatch('loadCity', responseData.home)
							.then(() => { context.commit('setLoaded', true) })
					} else if (data.cities && data.cities.length > 0) {
						// load the first city, if no home city is set
						context.dispatch('loadCity', data.cities[0].id)
					}
				}
			} catch (reason) {
				showError(context.getters.fatalError)
				logger.error('Cannot load cities', { reason })
			}
		},
		async loadCity(context, cityId) {
			const cityToLoad = context.state.allCities.cities.find(city => city.id === cityId)
			if (!cityToLoad) {
				return
			}

			try {
				const { data } = await axios.get(generateUrl('/apps/weather/weather/get'), {
					params: { name: cityToLoad.name }
				})
				const currentCityData = data
				currentCityData.wind.desc = windMapper(currentCityData.wind.deg)
				currentCityData.image = imageMapper(currentCityData.weather[0].main)

				context.commit('setCurrentCity', currentCityData)
				context.commit('setSelected', cityToLoad.id)
			} catch (reason) {
				showError(`Cannot load city: ${reason}`)
				if (reason.status === 404) {
					const error = t('weather', 'No city with this name found.')

					showError(error)
					context.commit('setCityLoadError', error)
					context.commit('setApiKeyNeeded', false)
				} else if (reason.status === 401) {
					const error = t(
						'weather',
						'Your OpenWeatherMap API key is invalid. Contact your administrator to configure a valid API key in Additional Settings of the Administration')
					showError(error)

					context.commit('setCityLoadError', error)
					context.commit('setApiKeyNeeded', true)
				} else {
					const error = context.getters.fatalError

					showError(error)

					logger.error('Error getting weather information for city', { reason })
					context.commit('setCityLoadError', error)
					context.commit('setApiKeyNeeded', false)
				}
			}
		},
		async deleteCity(context, cityId) {
			try {
				const { data } = await axios.post(generateUrl('/apps/weather/city/delete'), { id: cityId })
				if (data != null && !!data.deleted) {
					context.commit('removeCity', cityId)
					// If current city is the removed city, close it
					if (context.state.selectedCityId === cityId) {
						context.commit('unsetSelected')
						// if there are still cities on the list, and the selected was deleted, select the home city.
						// if the home city was deleted, select the first one instead
						if (context.state.allCities.cities.length > 0) {
							context.dispatch(
								'loadCity',
								cityId !== context.state.allCities.home
									? context.state.allCities.home
									: context.state.allCities.cities[0].id)
						}
					}
				} else {
					alert(t('weather', 'Failed to remove city. Please contact your administrator'))
				}
			} catch (reason) {
				showError(context.getters.fatalError)
				logger.error('Error deleting city', { reason })
			}
		},
		async setHome(context, cityId) {
			try {
				const { data } = await axios.post(generateUrl('/apps/weather/settings/home/set'), { city: cityId })
				if (data != null && !!data.set) {
					context.commit('setHome', cityId)
				} else {
					showError(t('weather', 'Failed to set home. Please contact your administrator'))
				}
				})
			} catch (reason) {
				logger.error('Error setting home city', { reason })
				showError(context.getters.fatalError)
			}
		},
		async selectCity(context, cityId) {
			await context.dispatch('loadCity', cityId)
			context.commit('setSelected', cityId)
		},
		async loadMetric(context) {
			try {
				const { data } = await axios.get(generateUrl('/apps/weather/settings/metric/get'))
				if (data.metric) {
					context.commit('setMetric', data.metric)
					context.commit('setMetricRepresentation', mapMetric(data.metric))
				}
			} catch (reason) {
				logger.error('Error setting metric', { reason })
				showError(context.getters.fatalError)
			}
		},
		async modifyMetric(context, metric) {
			try {
				const { data } = axios.post(generateUrl('/apps/weather/settings/metric/set'), { metric })
				if (data != null && !!(data.set)) {
					context.commit('setMetric', metric)
					context.commit('setMetricRepresentation', mapMetric(metric))
					context.dispatch('loadCity', context.state.selectedCityId)
				} else {
					showError(t('weather', 'Failed to set metric. Please contact your administrator'))
				}
			} catch (reason) {
				if (reason.status === 404) {
					showError(t('weather', 'This metric is not known.'))
				} else {
					logger.error('Error setting metric', { reason })
					showError(context.getters.fatalError)
				}
			}
		},
	},
```